### PR TITLE
Setup server callbacks

### PR DESCRIPTION
**Pull request by ClementBaziret**
This pull request implements the callbacks functions needed to send correct data to the clients

### CODE REVIEW

File: src/network/RTypeClient.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ void RTypeClient::engineInit()
     std::cout << "CompNb : " << (int)compNb << "." << std::endl;
     std::cout << "CompNameMaxSize : " << (int)compNameMaxSize << "." << std::endl;
 
-    for (uint16_t i; i < compNb; i++) {
+    for (uint16_t i = 0; i < compNb; i++) {
         std::vector<uint8_t> compName = blockingReceive();
         std::string strCompName = std::string(compName.begin() + 1, compName.end());
         // uint16_t CompId = (static_cast<uint16_t>(compName[0]) << 8) | static_cast<uint16_t>(compName[1]);
```
Comment (Tholarr) [inline src/network/RTypeClient.cpp:30]: comment left on purpose ?

File: src/network/UDPServer.hpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ class UDPServer {
         std::array<char, 1024> recv_buffer_;
         std::size_t size_max;
         std::unordered_map<std::string, std::type_index> __idStringToType;
+        // std::unordered_map<std::type_index, std::uint8_t> __newtorkCompIndex;
```
Comment (Tholarr) [inline src/network/UDPServer.hpp:42]: comment left on purpose ?

**Review by Tholarr — DISMISSED** [review]
Noice :+1:

File: src/network/RTypeClient.cpp (hunk 1 above)
Comment (ClementBaziret) [reply src/network/RTypeClient.cpp:30]: Yes It is for when we will switch to uint16 component type

File: src/network/UDPServer.hpp (hunk 2 above)
Comment (ClementBaziret) [reply src/network/UDPServer.hpp]: Nope, removed it

File: src/network/RTypeClient.cpp (hunk 1 above)
Comment (Tholarr) [reply src/network/RTypeClient.cpp:30]: alright :+1:

**Review by C0Florent — DISMISSED** [review]
Everything I had time to read looked good!

**Review by Tholarr — DISMISSED** [review]

**Review by Tholarr — APPROVED** [review]

**Review by C0Florent — APPROVED** [review]

**Pull request merged**